### ISSUE TITLE
unable to run default GCE network task

### ISSUE THREAD

**Issue by chrislovecnm** [kubernetes/kops#3678]
/area gce

User is reporting error with running network tasks on gce.  They are using master.

```console
I1020 15:58:43.335214   22722 changes.go:80] Field changed "Mode" actual="legacy" expected="auto"
W1020 15:58:43.335437   22722 executor.go:109] error running task "Network/default" (9m57s remaining to succeed): cannot apply changes to Network: *gcetasks.Network {"Name":null,"Lifecycle":null,"Mode":"auto","CIDR":null}
I1020 15:58:43.335491   22722 executor.go:91] Tasks: 34 done / 55 total; 1 can run
I1020 15:58:43.335518   22722 executor.go:157] Executing task "Network/default": *gcetasks.Network {"Name":"default","Lifecycle":"Sync","Mode":"auto","CIDR":null}
I1020 15:58:43.522245   22722 changes.go:80] Field changed "Mode" actual="legacy" expected="auto"
W1020 15:58:43.522530   22722 executor.go:109] error running task "Network/default" (9m57s remaining to succeed): cannot apply changes to Network: *gcetasks.Network {"Name":null,"Lifecycle":null,"Mode":"auto","CIDR":null}
I1020 15:58:43.522578   22722 executor.go:124] No progress made, sleeping before retrying 1 failed task(s)
I1020 15:58:53.522874   22722 executor.go:91] Tasks: 34 done / 55 total; 1 can run
I1020 15:58:53.522961   22722 executor.go:157] Executing task "Network/default": *gcetasks.Network {"Name":"default","Lifecycle":"Sync","Mode":"auto","CIDR":null}
I1020 15:58:53.680487   22722 changes.go:80] Field changed "Mode" actual="legacy" expected="auto"
```

**Comment by activeshadow** [comment]
Here's the cli command I used that generated the above errors.

```
$> export ZONES=us-central1-a,us-central1-b,us-central1-c
$> kops create cluster simple.k8s.local --zones $ZONES --master-zones $ZONES --node-count 10 --project ${PROJECT} --authorization RBAC --cloud gce --ssh-public-key ~/.ssh/id_ed25519.pub -v 10 --yes
```

**Comment by activeshadow** [comment]
If it helps any I get the same errors when I run the precompiled kops `v1.8.0-alpha.1 (git-991cd8c)`.

**Comment by chrislovecnm** [comment]
did you try getting a service account token?

**Comment by activeshadow** [comment]
I already had one. Didn’t try getting a new/different one.

**Comment by activeshadow** [comment]
Hi @chrislovecnm anything else I can try to get this working?

**Comment by chrislovecnm** [comment]
Reaching out to people, including @justinsb. Anyway to get a gist of that log, so more people can view?

**Comment by chrislovecnm** [comment]
Anything in https://cloud.google.com/compute/docs/activity-logs about the API call?

**Comment by activeshadow** [comment]
Here's a [gist](https://gist.github.com/activeshadow/88de07980d3de0bd296f504c0990e60f).

I'm not seeing anything in Stackdriver, though I'm not 100% sure what to filter for when it comes to network events.

I'm wondering if it has something to do with me trying to use the `default (legacy)` VPC network?

**Comment by bowei** [comment]
Is there a verbose mode to see what API error is actually causing this?

**Comment by activeshadow** [comment]
@bowei on the `kops` side of things, the gist I posted w/ `kops` logs was using `--v=10`.

**Comment by chrislovecnm** [comment]
Looking at this we are getting an odd error on the task validation.  @activeshadow 

1. `kops create cluster simple.k8s.local --zones $ZONES --master-zones $ZONES --node-count 10 --project ${PROJECT} --authorization RBAC --cloud gce --ssh-public-key ~/.ssh/id_ed25519.pub -v 10`
2. `kops get -o yaml`
3. `kops update cluster simple.k8s.local -v 10`
4. `kops update cluster simple.k8s.local -v 10 --yes`

That will break down the debugging some.  

@bowei I am not certain why the networking task in returning with different information.  https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/gcetasks/network.go is the code block, and it seems that the code is expecting legacy network mode but is getting auto subnet mode.

**Comment by activeshadow** [comment]
@chrislovecnm this [gist](https://gist.github.com/activeshadow/06c7bd1d62ac45fa54a12b6d943f6a48) has all the logs you requested above.

**Comment by chrislovecnm** [comment]
Is your netowork in auto or legacy mode. https://cloud.google.com/compute/docs/vpc/using-vpc

**Comment by activeshadow** [comment]
My `default` network is in `Legacy` mode.

**Comment by chrislovecnm** [comment]
Can you change it to auto?

**Comment by activeshadow** [comment]
No but I can create a new VPC that uses auto and try to use it for the cluster.

**Comment by activeshadow** [comment]
Any hints how to set the GCE VPC to use when creating the cluster? Using the `--vpc` option resulted in `error populating configuration: unable to find VPC ID "k8s-vpc"`.

**Comment by activeshadow** [comment]
@chrislovecnm @bowei is it possible to tell `kops` at cluster creation time what GCE VPC network to use, or can it only use the `default` VPC? Most of the VMs in my project are using the `default` VPC network, which is a `Legacy` network, and cannot be changed to `Auto`.

**Comment by chrislovecnm** [comment]
Can you try another region where you have auto enabled? I want to determine what the problem is, then how we fix it ;)

**Comment by activeshadow** [comment]
@chrislovecnm not sure what you mean by `try another region where you have auto enabled`... I have multiple VPC networks defined in my GCP project. The one named `default` is a legacy network. I have two other networks that are auto networks, and two others that are custom networks. The two auto networks include subnets in the `us-central1` region I'm trying to create a cluster in w/ `kops`.

**Comment by chrislovecnm** [comment]
I mac not certain how to set the vpc specifically. @justinsb?

**Comment by activeshadow** [comment]
Just wondering if there's been any movement on this behind the scenes? Anything I can try?

**Comment by activeshadow** [comment]
@chrislovecnm @justinsb just touching base on this again to see if anything new has happened. Please let me know how I can help... testing or otherwise.

**Comment by activeshadow** [comment]
@chrislovecnm @justinsb anything new on this? Anything I can test?

**Comment by activeshadow** [comment]
Just checked out `master`, rebuilt `kops` and tried again... same error. :frowning_face:

**Comment by justinsb** [comment]
Sorry about overlooking this issue for so long @activeshadow 

Just released kops 1.8.0.  We require that the Google network you're using not be in `legacy` mode - in fact we currently require it to be in `auto` mode.  You can change the mode of your network with `gcloud compute network change-mode`, or you can specify a different network entirely with `kops create cluster --vpc <name>` (which changes the cluster.spec.networkID value)

**Comment by bowei** [comment]
Migration from legacy to custom is not currently supported. You will have to create a new network.

From glcoud:
```
    gcloud compute networks switch-mode is used to change the mode of a
    network. Currently, only changing from auto to custom mode is supported.
```
